src/strkey.js

```diff
@@ -24,16 +32,16 @@ export function decodeCheck(versionByteName, encoded) {
     throw new Error(`invalid version byte. expected ${expectedVersion}, got ${versionByte}`);
   }
 
+  if (decoded.length !== 35) {
```

Comment:
I really don't like having a length check in this function...  the str-key format itself doesn't make any assumptions about the length of data being encoded or decoded.

We should validate lengths in either Account or (IMO) KeyPair.